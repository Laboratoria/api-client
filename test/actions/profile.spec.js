import {
  getComments,
  addComment,
  updateComment,
  deleteComment,
  getProfileTags,
  addProfileTag,
} from '../../lib/actions/profile';

describe('getComments', () => {
  it('should be a function', () => {
    expect(typeof getComments).toBe('function');
  });

  it('should create an action', () => {
    expect(getComments({ user: 'someone' })).toMatchSnapshot();
  });
});

describe('addComment', () => {
  it('should be a function', () => {
    expect(typeof addComment).toBe('function');
  });

  it('should create an action', () => {
    expect(addComment({
      user: 'someone',
      comment: {
        cohortProject: 'cohortProjectId',
        type: 'type',
        text: 'text',
      },
    })).toMatchSnapshot();
  });
});

describe('updateComment', () => {
  it('should be a function', () => {
    expect(typeof updateComment).toBe('function');
  });

  it('should create an action', () => {
    expect(updateComment({
      user: 'someone',
      comment: {
        id: 'commentId',
        text: 'text',
      },
    })).toMatchSnapshot();
  });
});

describe('deleteComment', () => {
  it('should be a function', () => {
    expect(typeof deleteComment).toBe('function');
  });

  it('should create an action', () => {
    expect(deleteComment({
      user: 'someone',
      commentId: 'commentId',
    })).toMatchSnapshot();
  });
});

describe('getProfileTags', () => {
  it('should be a function', () => {
    expect(typeof getProfileTags).toBe('function');
  });

  it('should create an action', () => {
    expect(getProfileTags({ user: 'someone' })).toMatchSnapshot();
  });
});

describe('addProfileTag', () => {
  it('should be a function', () => {
    expect(typeof addProfileTag).toBe('function');
  });

  it('should create an action', () => {
    expect(addProfileTag({
      user: 'someone',
      data: {
        assignmentReason: 'a comment',
        tag: 'tagId',
      },
    })).toMatchSnapshot();
  });
});
