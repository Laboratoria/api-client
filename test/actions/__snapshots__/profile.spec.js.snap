// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addComment should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "profile-comment-add",
    "namespace": "default",
    "suffix": "PROFILE_COMMENT_ADD",
  },
  "payload": Object {
    "data": Object {
      "cohortProject": "cohortProjectId",
      "text": "text",
      "type": "type",
    },
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/users/someone/profile/comments",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_COMMENT_ADD",
}
`;

exports[`addProfileTag should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "profile-tag-add",
    "namespace": "default",
    "suffix": "PROFILE_TAG_ADD",
  },
  "payload": Object {
    "data": Object {
      "assignmentReason": "a comment",
      "tag": "tagId",
    },
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/users/someone/profile/tags",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_TAG_ADD",
}
`;

exports[`deleteComment should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "profile-comment-delete",
    "namespace": "default",
    "suffix": "PROFILE_COMMENT_DELETE",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "DELETE",
    "params": undefined,
    "url": "/users/someone/profile/comments/commentId",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_COMMENT_DELETE",
}
`;

exports[`getComments should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": 300000,
    "key": "user/someone/profile/comments",
    "namespace": "default",
    "suffix": "PROFILE_COMMENTS",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/users/someone/profile/comments",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_COMMENTS",
}
`;

exports[`getProfileTags should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": 300000,
    "key": "user/someone/profile/tags",
    "namespace": "default",
    "suffix": "PROFILE_TAGS",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/users/someone/profile/tags",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_TAGS",
}
`;

exports[`updateComment should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "profile-comment-update",
    "namespace": "default",
    "suffix": "PROFILE_COMMENT_UPDATE",
  },
  "payload": Object {
    "data": Object {
      "text": "text",
    },
    "headers": undefined,
    "method": "PUT",
    "params": undefined,
    "url": "/users/someone/profile/comments/commentId",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_COMMENT_UPDATE",
}
`;

exports[`updateProfileTag should create an action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "profile-tag-update",
    "namespace": "default",
    "suffix": "PROFILE_TAG_UPDATE",
  },
  "payload": Object {
    "data": Object {
      "removalReasson": "a comment",
    },
    "headers": undefined,
    "method": "PUT",
    "params": undefined,
    "url": "/users/someone/profile/tags/tagId",
  },
  "type": "@@api-client/API_REQUEST/PROFILE_TAG_UPDATE",
}
`;
